python-checks/src/test/resources/checks/changeMethodContract.py

```diff
@@ -0,0 +1,65 @@
+class ParentClass(object):
+  def __init__(p1, p2): ...
+  def compliant(self, param1): ...
+  def my_method(self, param1): ...
+  def with_default(self, param1=42): ...
+  def changed_param_order(self, param1, param2): ...
+  def with_keyword_only(self, param1, *, param2): ...
+  def with_kwargs(self, param1, **kwargs): ...
+  @foo
+  def with_decorator(self, param1): ...
+  def __private_method(self): ...
+  attr = 42
+
+class ChildClass(ParentClass):
```

Comment:
I would also add a test for a case when a class inherits transitively from another.

Reply:
Test added in eb3d4366f4418598816482ff85941ff7650a314f